Extension/lib/settings/sync-service.js

```diff
@@ -534,11 +561,7 @@
         if (providerService.oauthSupported) {
             if ((!token || !api.oauthService.setToken(providerName, token, securityToken, expires))
                 && !api.oauthService.isAuthorized(providerName)) {
```

Comment:
I think we should move this logic from here. Set token directly via `oauthService`.